internal/server/http.go

```diff
@@ -47,43 +55,56 @@ func httpInit(group *run.Group, opts *options) error {
 	})
 
 	// Create our http server
-	httpSrv := &http.Server{
-		ReadHeaderTimeout: 5 * time.Second,
-		IdleTimeout:       120 * time.Second,
-		Handler:           httpLogHandler(rootHandler, log),
-		BaseContext: func(net.Listener) context.Context {
-			return opts.Context
+	return &httpServer{
+		opts: opts,
+		log:  log,
+		server: &http.Server{
+			ReadHeaderTimeout: 5 * time.Second,
+			IdleTimeout:       120 * time.Second,
+			Handler:           httpLogHandler(rootHandler, log),
+			BaseContext: func(net.Listener) context.Context {
+				return opts.Context
+			},
 		},
 	}
+}
 
-	// Add our gRPC server to the run group
-	group.Add(func() error {
-		// Serve traffic
-		ln := opts.HTTPListener
-		log.Info("starting HTTP server", "addr", ln.Addr().String())
-		return httpSrv.Serve(ln)
-	}, func(err error) {
-		ctx, cancelFunc := context.WithCancel(context.Background())
-		defer cancelFunc()
-
-		// Graceful in a goroutine so we can timeout
-		gracefulCh := make(chan struct{})
-		go func() {
-			defer close(gracefulCh)
-			log.Info("shutting down HTTP server")
-			httpSrv.Shutdown(ctx)
-		}()
+// start starts an http server
+func (s *httpServer) start() error {
+	// Serve traffic
+	ln := s.opts.HTTPListener
+	s.log.Info("starting HTTP server", "addr", ln.Addr().String())
+	return s.server.Serve(ln)
+}
 
-		select {
-		case <-gracefulCh:
+// close stops the grpc server, gracefully if possible.
+// Warning: before closing the GRPC server, this HTTP server must first be closed.
+// Attempting to gracefully stop the GRPC server first will cause it to drain HTTP connections,
+// which will panic.
+func (s *httpServer) close() {
+	log := s.log
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
 
-		// After a timeout we just forcibly exit. Our HTTP endpoints should
-		// be fairly quick and their operations are atomic so we just kill
-		// the connections after a few seconds.
-		case <-time.After(2 * time.Second):
-			cancelFunc()
+	// Graceful in a goroutine so we can timeout
+	gracefulCh := make(chan struct{})
+	go func() {
+		defer close(gracefulCh)
+		log.Info("stopping")
```

Comment:
```suggestion
		log.Debug("stopping")
```

Should this be `debug` like the grpc server and other logs here?